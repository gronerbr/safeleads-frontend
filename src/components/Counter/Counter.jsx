import React from 'react';
import PropTypes from 'prop-types';

const Counter = props => (
  <div>
    <h1>Total: {props.total}</h1>
    <button onClick={() => props.increment()}>Incrementar</button>
    <button onClick={() => props.decrement()}>Decrementar</button>
    <button onClick={() => props.decrementAsync()}>Decrementar ass√≠ncronamente</button>
  </div>
);

Counter.defaultProps = {
  increment: () => {},
  decrement: () => {},
  decrementAsync: () => {},
  total: 0,
};

Counter.propTypes = {
  decrement: PropTypes.func,
  increment: PropTypes.func,
  decrementAsync: PropTypes.func,
  total: PropTypes.number,
};

export default Counter;
