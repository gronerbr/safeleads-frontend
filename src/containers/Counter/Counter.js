import { connect } from 'react-redux';
import Counter from '../../components/Counter/Counter';
import { increment, decrement, decrementAsync } from '../../redux/modules/counter';

const mapStateToProps = state => ({
  total: state.counter.total,
});

const mapDispatchToProps = dispatch => ({
  increment: () => {
    dispatch(increment());
  },
  decrement: () => {
    dispatch(decrement());
  },
  decrementAsync: () => {
    dispatch(decrementAsync());
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(Counter);
