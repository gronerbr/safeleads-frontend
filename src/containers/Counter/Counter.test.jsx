import React from 'react';
import ReactDOM from 'react-dom';
import Counter from './Counter';
import createStore from '../../redux/create';

const store = createStore();

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Counter store={store} />, div);
});
