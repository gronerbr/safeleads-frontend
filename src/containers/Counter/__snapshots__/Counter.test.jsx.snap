// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Array [
  <h1>
    Total: 
    0
  </h1>,
  <button
    onClick={[Function]}
  >
    Incrementar
  </button>,
  <button
    onClick={[Function]}
  >
    Decrementar
  </button>,
  <button
    onClick={[Function]}
  >
    Decrementar ass√≠ncronamente
  </button>,
]
`;
